### ISSUE TITLE
Lambda layer invoke local --docker Breaking

### ISSUE THREAD

**Issue by michaelgutierrez0** [serverless/serverless#8936]
My team has multiple serverless framework projects. Within the provider section we are adding the layer. Inside gitignore we have .serverless. On some machines:
serverless invoke local --function get --path ./tests/getTestEvent.json --stage beta --docker
works and on others it does not. 

We noticed on machines that do not work the ./serverless/layers/lambda-layer-lib/1 is empty. We have copied over the .serverless contents from a working machine and tried putting them in git. When we run serverless invoke local it clears out that directory. We have tried docker with and without WSL. Have the latest version of serverless. Have tried deleting all images in docker to no avail. Have tried reinstalling docker. Tried using nodejs10.x and nodejs12.x but got the same error. Tried to use nodejs14.x which was added here: https://github.com/serverless/serverless/releases/tag/v2.23.0 and got:
failed to solve with frontend dockerfile.v0: failed to create LLB definition: docker.io/lambci/lambda:nodejs14.x: not found

<details>
<summary><code>serverless.yml</code></summary>

```
yaml
service: url-nerdifier

provider:
  name: aws
  runtime: nodejs12.x
  stage: ${opt:stage, env:PASSING_STAGE} # Use the command line option or if not provided use the PASSING_STAGE env var to populate the stage
  region: us-east-2 # Default Ohio region
  timeout: 5 # API Gateway timeout for integrations is 29 seconds
  memorySize: 512 # Default is 1024
  versionFunctions: false # Default is true, saves memory space
  role: arn:aws:iam::abc:role/lambda-cloudwatchlogs-vpc-sns-kms-secrets-write # Set the default role for all Lambda functions in this service
  environment:
    stage: ${opt:stage, env:PASSING_STAGE} # Create an environment variable for all Lambda functions named stage
  vpc:
    subnetIds:
      - subnet-abc
      - subnet-abc
      - subnet-abc
    securityGroupIds:
      - sg-076288e4263db62cb # Only allows traffic within the VPC and the MongoDB VPC peer
  layers:
    - arn:aws:lambda:us-east-2:abc:layer:lambda-layer-lib:1
  tags:
    stage: ${self:provider.stage} # Add a tag to track each stage's cost
    service: ${self:service} # Add a tag to track each services cost
functions:
  create:
    handler: src/handlers/create.handler
    name: ${self:service}-create-${self:provider.stage}
    description: "Creates a short URL given a long URL"
  get:
    handler: src/handlers/get.handler
    name: ${self:service}-get-${self:provider.stage}
    description: "Gets a long URL given a short URL"
    events:
      - http:
          path: /{proxy+}
          method: get
          cors: true
```

</details>

<details>
<summary><b><code>serverless invoke local --function get --path ./tests/getTestEvent.json --stage beta --docker</code> output</b></summary>

```
(node:24104) ExperimentalWarning: The fs.promises API is experimental
(node:24104) ExperimentalWarning: The dns.promises API is experimental
Serverless: To ensure safe major version upgrades ensure "frameworkVersion" setting in service configuration (recommended setup: "frameworkVersion: ^2.23.0")

Serverless: Load command interactiveCli
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command create
Serverless: Load command install
Serverless: Load command package
Serverless: Load command deploy
Serverless: Load command deploy:function
Serverless: Load command deploy:list
Serverless: Load command deploy:list:functions
Serverless: Load command invoke
Serverless: Load command invoke:local
Serverless: Load command info
Serverless: Load command logs
Serverless: Load command metrics
Serverless: Load command print
Serverless: Load command remove
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command slstats
Serverless: Load command plugin
Serverless: Load command plugin
Serverless: Load command plugin:install
Serverless: Load command plugin
Serverless: Load command plugin:uninstall
Serverless: Load command plugin
Serverless: Load command plugin:list
Serverless: Load command plugin
Serverless: Load command plugin:search
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command upgrade
Serverless: Load command uninstall
Serverless: Load command login
Serverless: Load command logout
Serverless: Load command generate-event
Serverless: Load command test
Serverless: Load command dashboard
Serverless: Load command output
Serverless: Load command output:get
Serverless: Load command output:list
Serverless: Load command param
Serverless: Load command param:get
Serverless: Load command param:list
Serverless: Load command studio
Serverless: Deprecation warning: Starting with next major version, API Gateway naming will be changed from "{stage}-{service}" to "{service}-{stage}".
            Set "provider.apiGateway.shouldStartNameWithService" to "true" to adapt to the new behavior now.
            More Info: https://www.serverless.com/framework/docs/deprecations/#AWS_API_GATEWAY_NAME_STARTING_WITH_SERVICE
Serverless: Invoke invoke:local
Serverless: Invoke package
Serverless: Invoke aws:common:validate
Serverless: Invoke aws:common:cleanupTempDir
Serverless: Packaging service...
Serverless: Excluding development dependencies...
Serverless: Invoke aws:package:finalize
Serverless: Invoke aws:common:moveArtifactsToPackage
2021-02-12T22:23:04.270Z        undefined       ERROR   Uncaught Exception      {"errorType":"Runtime.ImportModuleError","errorMessage":"Error: Cannot find module 'mongoose'\nRequire stack:\n- /var/task/src/models/HashItem.js\n- /var/task/src/models/index.js\n- /var/task/src/handlers/get.js\n- /var/runtime/UserFunction.js\n- /var/runtime/index.js","stack":["Runtime.ImportModuleError: Error: Cannot find module 'mongoose'","Require stack:","- /var/task/src/models/HashItem.js","- /var/task/src/models/index.js","- /var/task/src/handlers/get.js","- /var/runtime/UserFunction.js","- /var/runtime/index.js","    at _loadUserApp (/var/runtime/UserFunction.js:100:13)","    at Object.module.exports.load (/var/runtime/UserFunction.js:140:17)","    at Object.<anonymous> (/var/runtime/index.js:43:30)","    at Module._compile (internal/modules/cjs/loader.js:999:30)","    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)","    at Module.load (internal/modules/cjs/loader.js:863:32)","    at Function.Module._load (internal/modules/cjs/loader.js:708:14)","    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)","    at internal/main/run_main_module.js:17:47"]}
START RequestId: d0cbf83e-0c46-175b-8433-35b9a63c2105 Version: $LATEST
END RequestId: d0cbf83e-0c46-175b-8433-35b9a63c2105
REPORT RequestId: d0cbf83e-0c46-175b-8433-35b9a63c2105  Init Duration: 186.45 ms        Duration: 1.40 ms       Billed Duration: 2 ms   Memory Size: 512 MB     Max Memory Used: 52 MB

{"errorType":"Runtime.ImportModuleError","errorMessage":"Error: Cannot find module 'mongoose'\nRequire stack:\n- /var/task/src/models/HashItem.js\n- /var/task/src/models/index.js\n- /var/task/src/handlers/get.js\n- /var/runtime/UserFunction.js\n- /var/runtime/index.js"}

  Error --------------------------------------------------

  Error: Failed to run docker for nodejs12.x image (exit code 1})
      at AwsInvokeLocal.invokeLocalDocker (C:\Users\hurlt\AppData\Roaming\npm\node_modules\serverless\lib\plugins\aws\invokeLocal\index.js:508:13)
      at process._tickCallback (internal/process/next_tick.js:68:7)

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com

  Your Environment Information ---------------------------
     Operating System:          win32
     Node Version:              10.15.3
     Framework Version:         2.23.0
     Plugin Version:            4.4.3
     SDK Version:               2.3.2
     Components Version:        3.6.2
```

</details>

<b>Installed version</b>

```
Framework Core: 2.23.0
Plugin: 4.4.3
SDK: 2.3.2
Components: 3.6.2
```

**Comment by michaelgutierrez0** [comment]
Hi, Thanks for the response! Answers are below:
1. I am getting the same issue when not using docker. I was under the impression that local invokes of layers only worked if you used the --docker flag. Is that not correct?
2. I deleted: \.serverless\layers in the project directory and C:\Users\Username\AppData\Local\serverless\Cache\InvokeLocal. It then worked! The AppData directory is what did it I believe! I did not know it existed until a colleague pointed it out. Thanks for the help!
